lib/datadog/core/configuration/components.rb

```diff
@@ -75,6 +77,7 @@ def build_tracer(settings, agent_settings)
               enabled: settings.tracing.enabled,
               trace_flush: trace_flush,
               sampler: sampler,
+              span_sampler: build_span_sampler(settings),
```

Comment:
Don't love that `span_sampler` is its own component within the tracer, but it seems better to have smaller, simpler responsibilities than to have one sampler that's complex.